app/controllers/application_controller.rb

```diff
@@ -228,14 +230,22 @@ def prepare_catalog_index_vars(eq_models = nil)
     source_reservations =
       Reservation.not_returned.where(equipment_model_id: id_array).all
 
-    # build the hash using class methods that use 0 queries
     eq_models.each do |em|
+      # build the hash using class methods that use 0 queries
       @availability_hash[em.id] =
         [EquipmentObject.for_eq_model(em.id, eq_objects)\
         - Reservation.number_overdue_for_eq_model(em.id, source_reservations)\
         - em.num_reserved(cart.start_date, cart.due_date, source_reservations)\
         - cart.items[em.id].to_i, 0].max
+
+      # have requirements as part of equipment model itself
+      restricted = em.model_restricted?(cart.reserver_id)
+      if restricted
+        @qualifications_hash[em.id] = Requirement.list_requirement_admins(
+          User.find(cart.reserver_id), em).html_safe
```

Comment:
You're calling `User.find(cart.reserver_id)` for each restricted equipment model. I believe that ActiveRecord will cache the result, but you're probably better off storing the reserver outside of the `each` block (e.g. `reserver = User.find(cart.reserver_id)`).